### PR TITLE
[loss] Allow no loss in the model @open sesame 02/19 17:43

### PR DESCRIPTION
**Pull request by kparichay**
Allow setting no loss in the model
This allows inferencing a model, and creating submodels
with backbones to infer some particular output features

Updated existing unittest with no loss to succeed
and added more unittests to validate that models run successfully
without loss

See Also #927

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/utils/parse_util.cpp
Comment (lhs8928) [inline nntrainer/utils/parse_util.cpp:140]: 1. I think ll never goes to empty even the loss is omitted. 
Instead it would be a "none".
Check line 62 at model_loader.cpp

2. Also would it be better to declare a none at the beginning in loss_string, LossType?
Then these 2 line could be deleted :)

File: test/unittest/unittest_nntrainer_models.cpp
Comment (zhoonit) [inline test/unittest/unittest_nntrainer_models.cpp:492]: IMO, `*_vallidate` should have different test suite than `model_test` to keep consistency and reduce complexity.

How about we have this validation test added to `unittest_nntrainer_modelfile`? I think this is good spot because other modelfile should have a simplerun test as well.

This isn't directly related to PR. I am willing to work on it after this is merged.

**Review by zhoonit — APPROVED** [review]
LGTM

**Comment by zhoonit** [comment]
Please rebase this, there was a change regarding api dependency.

File: test/unittest/unittest_nntrainer_models.cpp
Comment (kparichay) [reply test/unittest/unittest_nntrainer_models.cpp:492]: A different test suite can be created. However, then each test suite will have to create its own models. If you planning to update test-suite, it would be good if unittests are able to merge and use each other defined models.

The reason for adding the tests here was to validate for now and then later run the same model.
Given this is a valid model, this test wants to check the training/inferencing this model as a user would do it (later add checking exact values).

> How about we have this validation test added to unittest_nntrainer_modelfile? I think this is good spot because other modelfile should have a simplerun test as well.

`unittest_nntrainer_modelfile` provides initialization check only.

IMO these tests can be done later.

File: nntrainer/utils/parse_util.cpp
Comment (kparichay) [reply nntrainer/utils/parse_util.cpp:140]: will do.

File: test/unittest/unittest_nntrainer_models.cpp
Comment (zhoonit) [reply test/unittest/unittest_nntrainer_models.cpp:492]: Understood, if that's the intention I think it explains good reason to leave `*_validate` here

**Review by zhoonit — APPROVED** [review]
LGTM after CI pass

File: nntrainer/utils/parse_util.cpp
Comment (kparichay) [reply nntrainer/utils/parse_util.cpp:141]: removed the empty check. `none` is already part of `loss_string`.

**Pull request merged**
